brick-editor.js

```diff
@@ -63,19 +75,43 @@ function addBlock(word) {
 
     // initialize dictionary
     var blockDict = {
-        'var': "var variableName = value;",
-        'if': "if (i == true) {\n" + tabs + "\t" + "// do something \n" + tabs + "}",
-        'if-else': "if (i == true) {\n" + tabs + "\t" + "// do something \n" + tabs + "} else {\n" + tabs + "\t" + "// do something \n" + tabs + "}",
-        'for': "for (var i = 0; i < value; i++){\n" + tabs + "\t // do something \n" + tabs + "}",
-        'while': "while (i < 10) {\n" + tabs + "\t" + "// do something \n" + tabs + "}",
-        'function': "function name(parameters) {\n" + tabs + "\t // do something \n" + tabs + "\t return value;\n" + tabs + "}",
+        'VARIABLE': "var variableName = value;",
+        'IF': "if (i == true) {\n" + tabs + "\t" + "// do something \n" + tabs + "}",
+        'IF-ELSE': "if (i == true) {\n" + tabs + "\t" + "// do something \n" + tabs + "} else {\n" + tabs + "\t" + "// do something \n" + tabs + "}",
+        'FOR': "for (var i = 0; i < value; i++){\n" + tabs + "\t // do something \n" + tabs + "}",
+        'WHILE': "while (i < 10) {\n" + tabs + "\t" + "// do something \n" + tabs + "}",
+        'FUNCTION': "function name(parameters) {\n" + tabs + "\t // do something \n" + tabs + "\t return value;\n" + tabs + "}",
     };
 
     var block = [firstPart, blockDict[word], lastPart].join("");
     editor.setValue(block);
     editor.setPosition(position);
+
 }
 
+// adds all the blocks to the button container
+function addBlocksHTML() {
+    for (i in keysArray) {
```

Comment:
This loop will have to change to accommodate the dictionary changes above. Using `for...in` in Javascript is also a bad idea. See https://stackoverflow.com/a/9329476

Reply:
Yes, I agree with this approach.